docs/api-reference.md

```diff
@@ -608,4 +624,140 @@ end
 As with `setState`, you can set use the constant `Roact.None` to remove a field from the state.
 
 !!! note
-	`getDerivedStateFromProps` is a *static* lifecycle method. It does not have access to `self`, and must be a pure function.
\ No newline at end of file
+	`getDerivedStateFromProps` is a *static* lifecycle method. It does not have access to `self`, and must be a pure function.
+
+---
+
+## ShallowWrapper
+
+### Fields
+
+#### type
+```
+type: {
+	kind: ElementKind
+}
+```
+
+The type dictionary always has the `kind` field that tell the component type. Additionally, depending of the kind of component, other information can be included.
+
+| kind | fields | description |
+| --- | --- | --- |
+| Host | className: string | the ClassName of the instance |
+| Function | functionComponent: function | the function that renders the element |
+| Stateful | component: table | the class-like table used to render the element |
+
+---
+
+#### props
+The props of the ShallowWrapper.
+
+!!! note
+	This dictionary will not contain the `Roact.Children` prop. To obtain the children elements wrapped into a ShallowWrapper, use the method `getChildren()`
+
+---
+
+#### hostKey
+The `hostKey` that is used to map the element to it's parent.
+
+---
+
+### Methods
+
+#### childrenCount
+```
+childrenCount() -> number
+```
+Returns the amount of children that the current ShallowWrapper contains.
+
+---
+
+#### find
+```
+find([constraints]) -> list[ShallowWrapper]
+```
+When a dictionary of constraints is provided, the function will filter the children that do not satisfy all given constraints. Otherwise, as `getChildren` do, it returns a list of all children wrapped into ShallowWrappers.
+
+---
+
+#### findUnique
+```
+findUnique([constraints]) -> ShallowWrapper
+```
+Similar to `find`, this method will assert that only one child satisfies the given constraints, or in the case where none is provided, will assert that there is simply only one child.
+
+---
+
+#### getChildren
+```
+getChildren() -> list[ShallowWrapper]
+```
+Returns a list of all children wrapped into ShallowWrappers.
+
+---
+
+#### getInstance
+```
+getInstance() -> Instance or nil
```

Comment:
Do we need getInstance? tostring + matchsnapshot + supporting methods to query the public fields of the shallow tree should be enough

Reply:
For more robust tests it can be interesting to directly assert on the instance. For example you can assert for a specific value of AbsoluteSize or AbsolutePosition. It could also useful if you want to test behavior from lifecycle methods that act on a ref.

Reply:
In such a test, could you just use a ref?

Reply:
If there is already a Ref, yes because you can access it through the props. Otherwise I don't think so